### PR TITLE
ESLint improvements

### PR DESCRIPTION
**Pull request by mondaychen**
I've upgraded all eslint related packages, and then applied all the automatic fixes that eslint suggests.

Currently, we still have 375 problems (375 errors, 0 warnings).

Sort by error type, we have:

```
import/no-extraneous-dependencies:                 2|
import/no-named-as-default:                        1|
import/no-named-default:                           3|
jsx-a11y/click-events-have-key-events:             4|
jsx-a11y/media-has-caption:                        1|
jsx-a11y/no-noninteractive-element-interactions:   2|
jsx-a11y/no-noninteractive-tabindex:               1|
jsx-a11y/no-static-element-interactions:           3|
max-len:                                           1|
no-case-declarations:                              1|
no-multi-assign:                                   2|
no-restricted-globals:                             1|
no-undef:                                          1|
no-unused-vars:                                    4|
prefer-destructuring:                             10|
prefer-rest-params:                                1|
react/default-props-match-prop-types:              1|
react/destructuring-assignment:                   80|
react/forbid-prop-types:                          44|
react/no-access-state-in-setstate:                 1|
react/no-find-dom-node:                            3|
react/no-this-in-sfc:                              1|
react/no-unused-prop-types:                       33|
react/no-unused-state:                             1|
react/require-default-props:                     172|
react/sort-comp:                                   1|
```

So I'm considering that we can disable `react/require-default-props` `react/destructuring-assignment` `react/forbid-prop-types` and `react/no-unused-prop-types` for now. Then we will be able to fix all other errors. In the future, we can discuss which of these we want to turn on again, and then take care of them one by one.

### CODE REVIEW

**Comment by mondaychen** [comment]
Update: I'll disable a few jsx-a11y rules as well since they take more effort to fix. Will track the accessibility improvement with a separate issue.

File: src/components/control-bar/LoadProgressBar.js [hunk 1]
```diff
@@ -60,7 +60,10 @@ export default function LoadProgressBar({ buffered, duration, className }) {
         className
       )}
     >
-      <span className="video-react-control-text"><span>Loaded</span>: 0%</span>
+      <span className="video-react-control-text">
+        <span>Loaded</span>
+: 0%
```
[the remark this answers is not in the document]
Comment (JimLiu) [reply src/components/control-bar/LoadProgressBar.js]: Not 100% sure with that, maybe we have a style `.video-react-control-text span`, I think you can remove the span tag

[the remark this answers is not in the document]
Comment (mondaychen) [reply src/components/control-bar/LoadProgressBar.js]: But `.video-react-control-text` is hidden itself

[the remark this answers is not in the document]
Comment (mondaychen) [reply src/components/control-bar/LoadProgressBar.js]: The only css file I found that contains this class name is progress.scss.

```
// Hide control text visually, but have it available for screenreaders
.video-react .video-react-control-text {
  @include hide-visually;
}
```

[the remark this answers is not in the document]
Comment (JimLiu) [reply src/components/control-bar/LoadProgressBar.js]: You are good to remove then 👍

**Comment by mondaychen** [comment]
@JimLiu let me know if you have further comments.

**Pull request merged**